src/main/java/com/faforever/client/update/ClientConfiguration.java

```diff
@@ -55,4 +58,9 @@
     private String message;
     private URL releaseNotesUrl;
   }
+
+  @Data
+  public static class OAuth {
+    private Collection<URI> redirectUris;
```

Comment:
Shouldn't these be apart of the ServerEndpoints object as the could be different depending on the endpoint used

Reply:
I also thought about this but it's a bit tricky; the redirect URIs are not server endpoints as in "where do I need to connect to". They are always `localhost` with some "random" port. But since the port must not be random but whitelisted by the OAuth server, the client needs to know which ports (or rather URIs), are whitelisted.

I added a JavaDoc.

Reply:
Yes but in terms of configuration these ports could be different between the test and production server so having a single value across both might not work

Reply:
That is true, but how would moving it to `ServerEndpoints` fix this?